### PR TITLE
 `joinRealtimeChannel` method with webxdc realtime API 

### PR DESCRIPTION
**Pull request by hpk42**
alternative to #77 implementing all realtime functiuonality through a single `joinRealtimeChannel` object.

### CODE REVIEW

File: src-docs/spec/joinRealtimeSession.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+return a `realtimeSession` object with `send` and `leave` methods. 
+
+The callback receives `Uint8Array` data items that were sent from connected peers. 
+
+Calling `joinRealtimeSession` while another `realtimeSession` is active
+will not cause any action and returns a `null` value. 
+
+Any transmitted realtime data is 
+
+- **private to the chat**: Only chat members can receive realtime data
```
[the remark this answers is not in the document]
Comment (hpk42) [reply src-docs/spec/joinRealtimeSession.md]: reformulated it. please check again.

File: src-docs/spec/joinRealtimeSession.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  and data can only be sent to connected chat members. 
+
+- **scoped to the app**: different apps in a
+  chat can not discover or receive realtime data of other apps in the chat. 
+
+- **ephemeral**: any sent data will only be received by the currently
+  connected chat peers but not by peers joining later.
```
[the remark this answers is not in the document]
Comment (hpk42) [reply src-docs/spec/joinRealtimeSession.md]: clarified it.

File: src-docs/spec/joinRealtimeSession.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+Send a `Uint8Array` data item to connected peers. 
+There is no guarantee anyone is receiving sent data
+because there might be no currently listening peers,
+or network connections fail. 
+It is up to the app to determine connectivity status with other peers. 
+
+## `realtimeSession.leave()`
+
+Leave the realtime session and disconnect from all peers.
```
[the remark this answers is not in the document]
Comment (hpk42) [reply src-docs/spec/joinRealtimeSession.md]: modified the phrasing accordingly.

File: src-docs/spec/joinRealtimeSession.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  connected peers but not by peers connecting later.
+
+Calling `joinRealtimeChannel` a second time without leaving the prior one
+will not cause any action and returns a `null` value. 
+
+## `realtimeChannel.setListener((data) => {})` 
+
+Start listening on the realtime channel using the specified callback. 
+The callback receives `Uint8Array` data items that were sent from connected peers. 
+Calling `setListener` a second time will replace the previous listener.
```
Comment (ansuz) [inline src-docs/spec/joinRealtimeSession.md:25]: what happens if the passed value is something other than a function? for instance:

```js
realtimeChannel.setListener()
```

Will that throw an error, or cause it to stop listening?

Comment (Septias) [reply src-docs/spec/joinRealtimeSession.md:25]: Will do so

File: src-docs/SUMMARY.md [hunk 5]
```diff
@@ -10,8 +10,8 @@
       - [sendToChat](./spec/sendToChat.md)
       - [importFiles](./spec/importFiles.md)
       - [selfAddr & selfName](./spec/selfAddr_and_selfName.md)
+      - [joinRealtimeSession](./spec/joinRealtimeSession.md)
```
[the remark this answers is not in the document]
Comment (hpk42) [reply src-docs/SUMMARY.md]: done

**Comment by Septias** [comment]
Should we merge this @hpk42 I think it is stable now

**Pull request merged**